### PR TITLE
Replace JSDOM with cheerio for scraping ACN

### LINKED ISSUE #78: Refactor: Replace JSDOM with Cheerio for web scraping
[JSDOM](https://github.com/jsdom/jsdom) is a whopping 2.4m (gzipped: 588.6 k) in size. It's only used for web scraping in this project, so it's a bit of an overkill and not worth it imo. [Cheerio](https://github.com/cheeriojs/cheerio) seems like the best lightweight alternative with the only drawback being the syntax from what I've seen so far.

Task:
- [ ] Replace jsdom with cheerio web scraping
- [ ] Uninstall jsdom

### PR DESCRIPTION
**Pull request by shaheer-2000**
## 🛠️ Fixes issue(s)

Closes #78 

## 👨‍💻 Change(s) proposed

- Introduces `cheerio` for web-scraping
- Uninstall's `jsdom`
- Uses `nodejs`' core `http` API to fetch `HTML` from ACN, and reduces additional package overhead
- Refactors `getTopics` and `getProjects` into a single `getAllResources` method

## ✔️ Checklist (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## 📄 Note to reviewers

There were no tests setup for this, nor any linting rules defined, hence the small change & push cycle.

### CODE REVIEW

File: backend/src/web-scrape-resources.js [hunk 1]
```diff
@@ -1,49 +1,75 @@
...
+    return [];
+  }
+
+  const $ = cheerio.load(html);
+
+  const resources = [];
+  const resourceTypes = ["Topics", "Projects"];
+
+  resourceTypes.forEach(resourceType => {
+    const _ = $("#sidebar").find(`li[title=${resourceType}] ul li a`);
```
Comment (Ngoakor12) [inline backend/src/web-scrape-resources.js]: what does `_` mean? can you please give it a descriptive name?

File: backend/src/web-scrape-resources.js [hunk 2]
```diff
@@ -1,49 +1,75 @@
...
+    _.each(function (i, el) {
+      resources.push({
+        title: $(this).text().replace(/\n/g, "").trim(),
+        // remove the preceding `/` from `href`
+        url: `${ACN_URL}${$(this).attr("href").slice(1)}`,
+        // Assuming the `resourceType` is a singular lowercased version of itself
+        type: resourceType.toLowerCase().slice(0, -1)
+      });
+    });
```
Comment (Ngoakor12) [inline backend/src/web-scrape-resources.js]: - Is it not possible to use a normal/common way of iterating? never used `.each` but it looks out of place.
- If not is it not possible to use an array function for callbacks instead?

**Review by Ngoakor12 — CHANGES_REQUESTED** [review]
Damn it I dislike the "jquery-like" syntax :sweat_smile: 

- Good call on turning the two topics and projects function into one, I think I like it better that way, plus I don't think I'll ever store them separately. The type property differentiates between the two well :ok_hand: 
- What are you using for documenting the functions? jsdoc? can I get a link? been meaning to learn some ways of documenting my code.
- I'm also not sure about using node's http since I don't have any experience with it, but I guess it works :man_shrugging:

**Comment by shaheer-2000** [comment]
> Damn it I dislike the "jquery-like" syntax 😅
> 
> * Good call on turning the two topics and projects function into one, I think I like it better that way, plus I don't think I'll ever store them separately. The type property differentiates between the two well 👌
> * What are you using for documenting the functions? jsdoc? can I get a link? been meaning to learn some ways of documenting my code.
> * I'm also not sure about using node's http since I don't have any experience with it, but I guess it works 🤷‍♂️

- Here's the reference link for jsdoc https://jsdoc.app/index.html 
- `cheerio` has a *jquery-ish* syntax 😅 
- Since it's a simple document fetch, incurring any additional overhead of using `axios` or something like it would be a bad decision. `cheerio` on the other hand is only responsible for parsing the document, not fetching it

File: backend/src/web-scrape-resources.js (hunk 1 above)
Comment (shaheer-2000) [reply backend/src/web-scrape-resources.js]: It was essentially a useless variable, returned a `Cheerio` instance. Could rename it to `targetResourceElements` or just `resourceElements`.

File: backend/src/web-scrape-resources.js (hunk 2 above)
Comment (shaheer-2000) [reply backend/src/web-scrape-resources.js]: `.each` is a method available on a `Cheerio` instance, it's the preferred way of traversing. Common JS Array methods, `.forEach` or `.map` aren't applicable here.

Reference: [https://cheerio.js.org/classes/Cheerio.html#each](https://cheerio.js.org/classes/Cheerio.html#each)

File: backend/src/web-scrape-resources.js (hunk 1 above)
Comment (Ngoakor12) [reply backend/src/web-scrape-resources.js]: That definitely seems sounds better than just a random underscore :+1:

File: backend/src/web-scrape-resources.js (hunk 2 above)
Comment (Ngoakor12) [reply backend/src/web-scrape-resources.js]: variables "i", "el" don't seem to be used anywhere in the code, can you remove them please

**Comment by Ngoakor12** [comment]
> * Here's the reference link for jsdoc [jsdoc.app/index.html](https://jsdoc.app/index.html)

thanks
> * `cheerio` has a _jquery-ish_ syntax sweat_smile

:sweat_smile: 
> * Since it's a simple document fetch, incurring any additional overhead of using `axios` or something like it would be a bad 
decision. `cheerio` on the other hand is only responsible for parsing the document, not fetching it

no problem

**Comment by shaheer-2000** [comment]
[0646133](https://github.com/Ngoakor12/coding-resource-finder/pull/155/commits/0646133c9f17368473bcd9207515e5a1ceedc64b) should resolve all of the above mentioned issues 😄

**Review by Ngoakor12 — APPROVED** [review]
Better :)

**Pull request merged**